### ISSUE TITLE
Bot transferring Higher IV's than threshold

### ISSUE THREAD

**Issue by leofe4** [jabbink/PokemonGoBot#1296]
### Prerequisites (Remove this section if you want)
Make sure you...

* Have checked the FAQ if your question has not been answered there.
* Have searched the existing issues if your question has not been asked here.
* Post a FULL stacktrace and not only a part of it.
* Fill in all the fields correctly, it helps us immensly if we get a complete image of your situation.

**Description:**
I set my my IV threshold at 80 and it transferred a bunch of my pokemon that were higher than this:

# Minimum amount of pokemon type to keep
keep_pokemon_amount=0

# Maximum amount of pokemon type to keep (-1 to keep all)
max_pokemon_amount=0

# Sort by IV first instead of CP
sort_by_iv=true

# Minimum IV percentage to keep a pokemon (to ignore IV: use -1)
# between 0 and 100, suggested 80
transfer_iv_threshold=80

# Minimum CP to keep a pokemon (to ignore CP: use -1)
transfer_cp_threshold=-1

# Minimum CP % in relation to max CP of pokemon to your current trainer lvl to keep pokemon (to disable CP%:use -1)
# e.g. Trainer lvl is 33, captured EEVEE has 490 CP which results in max possible CP of 984 with 15/14/15 IV
# that however is below the threshold of 60% because its 49%
transfer_cp_min_threshold=40

**Steps to reproduce:**

1. [Step 1]
2. [Step 2]

**Expected behavior:**
Keep pokemon with IVs greater than 80

**Actual behavior:**
Transferred Pokemon with IVs greater than 80

Going to transfer MAGIKARP with CP 143 and IV 91%; reason: Too many

Going to transfer ARCANINE with CP 1841 and IV 84%; reason: Too many

Going to transfer BEEDRILL with CP 796 and IV 84%; reason: Too many

Going to transfer GOLDEEN with CP 401 and IV 86%; reason: Too many

**Stacktrace (If it's a crash):**
[Please use pastebin if it's too long]

**Version:**
PokemonGoBot-0.6.0

**Operating System:**
Windows 10

**Java version:**
Version 8 update 60 (buld 1.8.0_60-b27

Leave this sentence in your issue as proof that you have read and used this template.

**Comment by Cao-Lao** [comment]
try changing that :
max_pokemon_amount=0
into that :
max_pokemon_amount=-1
or put a number instead

if you read carefully you can see that :
Going to transfer MAGIKARP with CP 143 and IV 91%; **reason: Too many**

**Comment by leofe4** [comment]
Yeah. I tried -1 and it for some reason kept on transferring. I placed a 5 there and it stopped transferring higher IVs. 

BTW they are sorted by IVs and not CPs

**Comment by digihax** [comment]
if you tell it to keep a max_pokemon_amount of zero -- it's going to transfer all of them.
Set that value to 1 or 2, and it will keep 1 or 2 *of each type* of pokemon, so 2 bulbasaurs, 2 ivysaurs, 2 venusaur's, etc.

Your bag of pokemon will quickly fill up if you set it to -1 (unlimited) or 3+, in my experience.  
I don't care about CP, I keep by IV, and my minimum threshold is 94% and above (so effectively, 95/97/100).  Since I keep 2 of each type, I will keep pokemon below that threshold to meet the minimum of two, but over time, they will be the *best two*. 

Now, since I also keep anything over 94%, I will possibly have 3 or 4 of some types, if their IVs are 97 or 100.  So I have for instance, 4 97% vaporeon's.  Sometimes I keep them all (for trading eventually), and I definitely keep those with different attack sets.

Anyway: hope that helps you sort out some of the settings and impact.  The bot's really good about giving you a reason why it transfers and that should guide you in how to update it to achieve what you want.

**Comment by Cao-Lao** [comment]
try deleting default.json
it's in bot setting folder

**Comment by leofe4** [comment]
My settings folder is empty already Cao. :(
